src/main/java/kotlinx/reflect/lite/KProperty.kt

```diff
@@ -0,0 +1,249 @@
+     *                 or an extension receiver if this is a top level extension property.
+     * @param value the new value to be assigned to this property.
+     */
+    public fun set(receiver: T, value: V)
+
+    override val setter: Setter<T, V>
+
+    /**
+     * Setter of the property is a `set` method declared alongside the property.
+     *
+     * Can be used as a function that takes the receiver and the new property value as arguments and returns [Unit].
+     */
+    public interface Setter<T, V> : KMutableProperty.Setter<V>, (T, V) -> Unit
+}
+
+
+/**
+ * Represents a property, operations on which take two receivers as parameters,
+ * such as an extension property declared in a class.
+ *
+ * @param D the type of the first receiver. In case of the extension property in a class this is
+ *        the type of the declaring class of the property, or any subclass of that class.
+ * @param E the type of the second receiver. In case of the extension property in a class this is
+ *        the type of the extension receiver.
+ * @param V the type of the property value.
+ */
+public interface KProperty2<D, E, out V> : KProperty<V>, (D, E) -> V {
+    /**
+     * Returns the current value of the property. In case of the extension property in a class,
+     * the instance of the class should be passed first and the instance of the extension receiver second.
+     *
+     * @param receiver1 the instance of the first receiver.
+     * @param receiver2 the instance of the second receiver.
+     */
+    public fun get(receiver1: D, receiver2: E): V
+
+    /**
+     * Returns the value of the delegate if this is a delegated property, or `null` if this property is not delegated.
+     * See the [Kotlin language documentation](https://kotlinlang.org/docs/reference/delegated-properties.html)
+     * for more information.
+     *
+     * In case of the extension property in a class, the instance of the class should be passed first
+     * and the instance of the extension receiver second.
+     *
+     * @param receiver1 the instance of the first receiver.
+     * @param receiver2 the instance of the second receiver.
+     *
+     * @see [kotlin.reflect.full.getExtensionDelegate] // [KProperty2.getExtensionDelegate]
+     */
+    @SinceKotlin("1.1")
+    private fun getDelegate(receiver1: D, receiver2: E): Any? = TODO()
+
+    override val getter: Getter<D, E, V>
+
+    /**
+     * Getter of the property is a `get` method declared alongside the property.
+     *
+     * Can be used as a function that takes an argument of type [D] (the first receiver), an argument of type [E] (the second receiver)
+     * and returns the value of the property type [V].
+     */
+    public interface Getter<D, E, out V> : KProperty.Getter<V>, (D, E) -> V
+}
+
+/**
+ * Represents a `var`-property, operations on which take two receivers as parameters.
+ */
+public interface KMutableProperty2<D, E, V> : KProperty2<D, E, V>, KMutableProperty<V> {
```

Comment:
This is not implemented by anything, so publishing this would be misleading: people would try to cast to it, but unsuccessfully, even when logically it should make sense. Ditto for the other such interfaces.